### DISCUSSION TITLE
Retrieving updated file source from org for a class then has unsaved changes prompt for merge or replace

### DISCUSSION THREAD

**Discussion by adayIvey** [forcedotcom/salesforcedx-vscode#6274] [Ideas]
If I updated a class(any file) from the org that has unsaved changes in vscode there is no way for me to know if the updated file i just pulled has changes that are not in my unsaved version and i may overwrite them when i save my local version.

It would be nice if there was a setting that could be enabled (like the diff check on deploy) that the file being retrieved is checked against my unsaved changes and if a diff is detected then i would be prompted if i wanted to replace (and lose my changes) or allow me to merge the difference into my unsaved class

**Comment by gbockus-sf** [comment]
Hi @aday12345 thanks for the feedback.  

So basically you're looking for the same conflict detection we have on  deploy but also for retrieves.  Seems logical to me.  @AnanyaJha Care to weigh in if this is on the road map anywhere?

**Comment by AnanyaJha** [comment]
Hi @aday12345 , thanks for surfacing this! We don't have this on our immediate roadmap, but we'll definitely keep this in mind for future enhancements to the Conflict Detection feature. To make sure I understand the use case correctly: _when you retrieve a file from the org_, are you looking to understand whether or not the retrieve operation will overwrite your local changes? or are you looking for some sort of verification _when you're saving a file locally_?

**Comment by adayIvey** [comment]
The retrieve operation is overwriting my local saved changes.

Say I have changed a class file on my local machine and saved it (but have not deployed to the sandbox).
Then I do a retrieve source on that file or folder etc. to get latest my locally saved changes gets overwritten with the file from the sandbox.

It would be nice if it was able to detect the differences and ask me if I want to overwrite or merge the two files.



BTW: The conflict auto detection on deploy is great and I would suggest that it should be enabled by default.

**Comment by AnanyaJha** [comment]
Thanks for clarifying @aday12345 ! We'll keep this feature work in mind for our roadmap.

**Comment by mshanemc** [comment]
conflict checking now works for retrieves

**Comment by adayIvey** [comment]
thanks :)
